src/hardware/serial_terminal.rs

```diff
@@ -0,0 +1,134 @@
+//! Booster NGFW Application
+use super::flash::Flash;
+use super::{platform, UsbBus};
+use crate::settings::global_settings::BoosterMainBoardData;
+use core::fmt::Write;
+use embedded_storage::nor_flash::NorFlash;
+
+pub struct SerialSettingsPlatform {
+    pub metadata: &'static crate::hardware::metadata::ApplicationMetadata,
+    pub settings: BoosterMainBoardData,
+    pub storage: Flash,
+
+    /// The interface to read/write data to/from serially (via text) to the user.
+    pub interface: serial_settings::BestEffortInterface<usbd_serial::SerialPort<'static, UsbBus>>,
+}
+
+#[derive(Debug)]
+pub enum Error<F> {
+    Postcard(postcard::Error),
+    Flash(F),
+}
+
+impl<F> From<postcard::Error> for Error<F> {
+    fn from(e: postcard::Error) -> Self {
+        Self::Postcard(e)
+    }
+}
+
+impl serial_settings::Platform for SerialSettingsPlatform {
+    type Interface = serial_settings::BestEffortInterface<usbd_serial::SerialPort<'static, UsbBus>>;
+
+    type Settings = crate::settings::global_settings::BoosterMainBoardData;
+
+    type Error = Error<<Flash as embedded_storage::nor_flash::ErrorType>::Error>;
+
+    fn save(&mut self, buffer: &mut [u8]) -> Result<(), Self::Error> {
+        let serialized = postcard::to_slice(self.settings(), buffer)?;
+        self.storage
+            .erase(0, serialized.len() as u32)
+            .map_err(Self::Error::Flash)?;
+        self.storage
+            .write(0, serialized)
+            .map_err(Self::Error::Flash)?;
+        Ok(())
+    }
+
+    /// Execute a platform specific command.
+    fn cmd(&mut self, cmd: &str) {
+        match cmd {
+            "reboot" => {
+                cortex_m::interrupt::disable();
+                platform::shutdown_channels();
```

Comment:
Where's the reboot?

Reply:
Woops! Indeed it was missing. I fixed it in f06e361677b0c133969694f001589ca9978d154b